src_emails/email_app.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+# -*- coding: UTF-8 -*-
+
+import os
+import random
+import time
+import datetime
+import click
+import logging
+# import time
+import json
+
+from flask import Flask, current_app, g, request, jsonify, abort
+
+
+import psycopg2.pool
+
+
+
+app = Flask(__name__)
+
+app.config['DB_POOL_COUNT_MIN'] = os.environ.get('DB_POOL_COUNT_MIN', 2)
+app.config['DB_POOL_COUNT_MAX'] = os.environ.get('DB_POOL_COUNT_MAX', 10)
+app.config['DB_POOL_GETCONN_ATTEMPTS'] = os.environ.get('DB_POOL_GETCONN_ATTEMPTS', 10)
+app.config['DATABASE'] = f"postgresql://{ os.environ['POSTGRESQL_USER'] }:{ os.environ['POSTGRESQL_PASSWORD'] }@{ os.environ['POSTGRESQL_HOST'] }:{ os.environ['POSTGRESQL_PORT'] }/{ os.environ['POSTGRESQL_DATABASE'] }"
+app.logger.setLevel(logging.INFO)  # Set log level to INFO
+handler = logging.FileHandler('app.log')  # Log to a file
+app.logger.addHandler(handler)
+
+
+##########
+# DB setup
+##########
+
+def get_db():
+    if 'db' not in g:
+        for attempt in range(current_app.config["DB_POOL_GETCONN_ATTEMPTS"]):
+            try:
+                g.db = current_app.config["db_pool"].getconn()
+            except:
+                time.sleep(.1)
+            else:
+                break
+        else:
+            raise Exception("Failed to get DB connection")
+
+    return g.db
+
+def close_db(e=None):
+    db = g.pop('db', None)
+
+    if db is not None:
+        current_app.config["db_pool"].putconn(db)
+
+
+app.teardown_appcontext(close_db)
+app.config["db_pool"] = psycopg2.pool.ThreadedConnectionPool(
+    app.config['DB_POOL_COUNT_MIN'],
+    app.config['DB_POOL_COUNT_MAX'],
+    app.config['DATABASE'],
+)
+app.logger.info(f"Initialized DB pool (min {app.config['DB_POOL_COUNT_MIN']}, max {app.config['DB_POOL_COUNT_MIN']})")
+
+
+##########
+# Routes
+##########
+
+
+@app.route('/code/success/<endpoint>', methods=['GET', 'POST'])
```

Comment:
I guess this is still WIP right?

Reply:
hi @jhutar yeah it is different. I wanted to test new service works. Actualy implementation is this one -> https://backoffice-proxy.apps.ext.spoke.preprod.us-west-2.aws.paas.redhat.com/docs/#/default/post_v1_sendEmails